src/interfaces.ts

```diff
@@ -162,3 +167,27 @@ export interface CommentFlag {
     affectedRange: Range;
     codes: number[] | null;
 }
+
+type ValidateHandler = (scope: Scope, files: (BrsFile | XmlFile)[], callables: CallableContainerMap) => void;
+
+export interface CompilerPlugin {
```

Comment:
I'd like to standardize the names so we're using the same tense for them all. Here's what I'm thinking:

- beforePrepublish
- beforePublish
- afterPublish
- afterProgramCreate
- beforeProgramValidate
- afterProgramValidate
- beforeScopeCreate
- afterScopeDispose
- beforeScopeValidate
- afterScopeValidate
- afterFileSourceLoad
- afterFileParse
- beforeFileValidate
- afterComponentSourceLoad
- afterComponentParse
- afterComponentValidate
- beforeTranspile
- afterTranspile

Reply:
I like it - actually making `afterFileSourceLoad` -> `beforeFileParse`